linera-base/src/crypto.rs

```diff
@@ -67,21 +67,18 @@ pub enum CryptoError {
 
 impl PublicKey {
     /// A fake public key used for testing.
-    #[cfg(any(test, feature = "test"))]
+    #[cfg(with_testing)]
     pub fn debug(name: u8) -> PublicKey {
         let addr = [name; dalek::PUBLIC_KEY_LENGTH];
         PublicKey(addr)
     }
 }
 
 /// Wrapper around [`rand07::CryptoRng`] and [`rand07::RngCore`].
-#[cfg(not(target_arch = "wasm32"))]
```

Comment:
This needs to be feature-gated somehow. It would actually compile in a smart contract, then crash at runtime.

Reply:
Oh I didn't realize `rand07` doesn't feature-gate this.

What's the reason for pinning an old version of `rand` here?

Reply:
Apropos of offline conversation: no reason except legacy.  Upgrading to `rand` 0.8 in https://github.com/linera-io/linera-protocol/pull/1678, then I will rebase this PR.

Reply:
(which will force `generate` to be feature-gated)

Reply:
Any reason not feature-gate the trait definition itself?

Reply:
Gated, though this re-introduces the `web` feature gate noise everywhere that I was trying to avoid by making crates Wasm-environment–agnostic.